src/classes/RD2_ScheduleService.cls

```diff
@@ -535,6 +534,84 @@ public without sharing class RD2_ScheduleService {
         return schedulesByRdId;
     }
 
+    public List<ActiveSchedule> getVisualizedSchedules(npe03__Recurring_Donation__c rd) {
+        List<ActiveSchedule> activeSchedules = new List<ActiveSchedule>();
+
+        if (rd == null) {
+            return activeSchedules;
+        }
+
+        List<RecurringDonationSchedule__c> schedules = getExistingSchedules(rd);
+        if (schedules == null) {
+            return activeSchedules;
+        }
+
+        Boolean first1stAnd15thFound = false;
+        for (RecurringDonationSchedule__c schedule : schedules) {
+
+            // This block collapses 1st and 15th schedules into a single row and nulls out day of month
+            if (schedule.InstallmentPeriod__c == RD2_Constants.INSTALLMENT_PERIOD_FIRST_AND_FIFTEENTH) {
+                if (first1stAnd15thFound) {
+                    first1stAnd15thFound = false;
+                    continue;
+                }
+                first1stAnd15thFound = true;
+                schedule.DayOfMonth__c = null;
```

Comment:
I don't understand how this would work to meet the need.  Can you clarify?

Reply:
What I think @RadSFDeveloper is suggesting here is that since this method is setting `schedule.DayOfMonth__c` to null specifically so that it's null when it gets to the ActiveSchedule inner class and its not used anywhere else in this method, that it makes sense to change DayOfMonth to null in the constructor when the `schedule.InstallmentPeriod__c == RD2_Constants.INSTALLMENT_PERIOD_FIRST_AND_FIFTEENTH`. I agree that it  encapsulates the logic (setting the field to null) to where its needed as null. The rest of the if block (i.e., to ignore the "first" `1st_and_15th` record) is fine in this method.